### PR TITLE
Remove not used feature flags

### LINKED ISSUE #46844: [Enhancement]: Remove not used feature flags
Remove legacy feature flags. There are many legacy feature flags related to the blocks editor that are not being used anymore, we want to remove them.

This is a list of featured flags that it seems are not being used anymore:

- product-variation-management
- product-virtual-downloadable
- product-external-affiliate
- product-grouped
- product-linked
- product-pre-publish-modal
- product-custom-fields

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR removes the following non-used feature flags

- product-variation-management
- product-virtual-downloadable
- product-external-affiliate
- product-grouped
- product-linked
- customer-effort-score-tracks

Closes #46844.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add and do some smoke tests to verify that anything is broken.
3. Try the variations, downloadable products, change the product type, add custom fields, publish a couple of products, etc.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/config/core.json [hunk 1]
```diff
@@ -19,13 +18,6 @@
 		"new-product-management-experience": false,
 		"onboarding": true,
 		"onboarding-tasks": true,
-		"product-variation-management": true,
-		"product-virtual-downloadable": true,
-		"product-external-affiliate": true,
-		"product-grouped": true,
-		"product-linked": true,
-		"product-pre-publish-modal": true,
-		"product-custom-fields": true,
```
Comment (louwie17) [inline plugins/woocommerce/client/admin/config/core.json]: I would keep at-least these two, as first of all we are actually planning on disabled the `pre-publish` feature https://github.com/woocommerce/woocommerce/issues/47189

Also the custom fields is still relatively new, so I think we should keep the feature flag incase something big comes up.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Request a change on a couple of the feature flags.

File: plugins/woocommerce/client/admin/config/core.json (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce/client/admin/config/core.json]: Good call! I just removed the commits that deleted those flags.

**Comment by octaedro** [comment]
Hey @louwie17, I just addressed the change you mentioned.
Could you take another look at this PR?

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Thanks for cleaning this up @octaedro

**Pull request merged**